src/uct/ib/dc/accel/dc_mlx5.c

```diff
@@ -553,8 +554,22 @@ static void uct_dc_mlx5_iface_progress(void *arg)
 static UCS_F_NOINLINE void uct_dc_mlx5_iface_handle_failure(uct_ib_iface_t *ib_iface,
                                                             void *arg)
 {
-    struct mlx5_cqe64 *cqe = arg;
-    uct_ib_mlx5_completion_with_err((void*)cqe, 0);
+    struct mlx5_cqe64   *cqe    = arg;
+    uint32_t            qp_num  = ntohl(cqe->sop_drop_qpn) & UCS_MASK(UCT_IB_QPN_ORDER);
+    uct_dc_mlx5_iface_t *iface  = ucs_derived_of(ib_iface, uct_dc_mlx5_iface_t);
+    uint8_t             dci     = uct_dc_iface_dci_find(&iface->super, qp_num);
+    uct_dc_ep_t         *ep     = iface->super.tx.dcis[dci].ep;
+
+    if (ep) {
+        uct_rc_txqp_purge_outstanding(&iface->super.tx.dcis[dci].txqp,
```

Comment:
can extract this code to common function for dc_mlx5 and dc_verbs
also, need to revive the dci since after the failure the failed dci becomes unusable